src/lxc/utils.c

```diff
@@ -1868,13 +1868,15 @@ int fix_stdio_permissions(uid_t uid)
 
 bool multiply_overflow(int64_t base, uint64_t mult, int64_t *res)
 {
-	if (base > 0 && base > (int64_t)(INT64_MAX / mult))
-		return false;
+	int64_t tmp;
+
+	tmp = base * mult;
 
-	if (base < 0 && base < (int64_t)(INT64_MIN / mult))
+	if (base == (tmp / (int64_t)mult)) {
```

Comment:
As far as I understand compiler can optimize-out this expression and decide that the whole condition is always true.

I would suggest easier fix, just replace `base < (int64_t)(INT64_MIN / mult)` with `base < -(int64_t)(INT64_MIN / mult)`. It takes into account that for any `mult`, always `(int64_t)(INT64_MIN / mult) > 0`. It's because `mult` is unsigned and wider than `int64_t`.

We can consider using `__builtin_mul_overflow`, but it's available only with GCC versions >= 5.

cc @stgraber @brauner

Reply:
GCC 5 has been around for almost 10 years so should be fine.

Reply:
Using `__builtin_mul_overflow` seems the best option.
Would it suits everyone if I replace the function body entirely with a single call to this builtin?

Reply:
> Would it suits everyone if I replace the function body entirely with a single call to this builtin?

we have special helper already (it's called `check_mul_overflow` and it uses this builtin function).

Dear Arnaud,

I would suggest you to wait a bit and I'll prepare a new PR with this fixed and I'll include your commit (with you authorship preserved ) as a part of this new PR.

The only question I have is that which email you want to use as a git author email? In your signed-off-by tag I see `151835536+af-airbus@users.noreply.github.com` which seems a bit weird github-specific email. Are you sure that you want to go with this address?

Reply:
Please, check https://github.com/lxc/lxc/pull/4418

I have added your commit and set you as an author of it (also added your signed-off-by tag)